[gettext] build failure with tools feature

**Host Environment**
- OS: Windows 10.0.19042.906
- Compiler: Visual Studio 2019

**To Reproduce**
Steps to reproduce the behavior:
`.\vcpkg.exe install --triplet x64-windows gettext`

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
```text
Computing installation plan...
The following packages will be built and installed:
    gettext[core]:x64-windows -> 0.21#1
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\wangqr\AppData\Local\vcpkg\archives\3f\3f89b73f0fe2bc0af7060cb548c1577802a70081.zip
Starting package 1/1: gettext:x64-windows
Building package gettext[core]:x64-windows...
-- Using cached C:/Users/wangqr/Files/src/vcpkg/downloads/gettext-0.21.tar.gz
-- Cleaning sources at C:/Users/wangqr/Files/src/vcpkg/buildtrees/gettext/src/0.21-508c311661.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/wangqr/Files/src/vcpkg/downloads/gettext-0.21.tar.gz
-- Applying patch 0002-Fix-uwp-build.patch
-- Applying patch 0003-Fix-win-unicode-paths.patch
-- Applying patch rel_path.patch
-- Using source at C:/Users/wangqr/Files/src/vcpkg/buildtrees/gettext/src/0.21-508c311661.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Using msys root at C:/Users/wangqr/Files/src/vcpkg/downloads/tools/msys2/f04a632dd84902ee
-- Using msys root at C:/Users/wangqr/Files/src/vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Using msys root at C:/Users/wangqr/Files/src/vcpkg/downloads/tools/msys2/969c0913b9df89e1
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: C:/Users/wangqr/Files/src/vcpkg/downloads/tools/msys2/f04a632dd84902ee/usr/bin/make.exe -j 13 --trace -f Makefile all
    Working Directory: C:/Users/wangqr/Files/src/vcpkg/buildtrees/gettext/x64-windows-dbg
    See logs for more information:
      C:\Users\wangqr\Files\src\vcpkg\buildtrees\gettext\build-x64-windows-dbg-out.log
      C:\Users\wangqr\Files\src\vcpkg\buildtrees\gettext\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:195 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/gettext/portfile.cmake:67 (vcpkg_install_make)
  scripts/ports.cmake:142 (include)


Error: Building package gettext:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gettext:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
```
-Please attach any additional failure logs mentioned in the console output.
Here is the `vcpkg\buildtrees\gettext` without `src\`:
[gettext.zip](https://github.com/microsoft/vcpkg/files/6291801/gettext.zip)

**Additional context**
The errors reported in `build-x64-windows-dbg-out.log`:
```text
../.././../src/0.21-508c311661.clean/gettext-tools/gnulib-lib/closedir.c(40): error C2084: function 'int closedir(DIR *)' already has a body
C:/Users/wangqr/Files/src/vcpkg/installed/x64-windows/include\dirent.h(838): note: see previous definition of 'closedir'
../.././../src/0.21-508c311661.clean/gettext-tools/gnulib-lib/closedir.c(58): error C2039: 'current': is not a member of 'DIR'
C:/Users/wangqr/Files/src/vcpkg/installed/x64-windows/include\dirent.h(275): note: see declaration of 'DIR'
../.././../src/0.21-508c311661.clean/gettext-tools/gnulib-lib/closedir.c(59): error C2039: 'current': is not a member of 'DIR'
C:/Users/wangqr/Files/src/vcpkg/installed/x64-windows/include\dirent.h(275): note: see declaration of 'DIR'
../.././../src/0.21-508c311661.clean/gettext-tools/gnulib-lib/closedir.c(59): error C2198: 'FindClose': too few arguments for call
```

--- a/ports/gettext/portfile.cmake
+++ b/ports/gettext/portfile.cmake
@@ -43,6 +43,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
                         gl_cv_func_printf_directive_n=no #segfaults otherwise with popup window
                         ac_cv_func_memset=yes #not detected in release builds 
                         ac_cv_header_pthread_h=no
+                        ac_cv_header_dirent_h=no
                         )
 endif()
 set(ADDITIONAL_CONFIGURE_OPTIONS)

--- a/ports/gettext/vcpkg.json
+++ b/ports/gettext/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "gettext",
   "version": "0.21",
-  "port-version": 1,
+  "port-version": 2,
   "description": "The GNU gettext utilities are a set of tools that provides a framework to help other GNU packages produce multi-lingual messages. Provides libintl.",
   "homepage": "https://www.gnu.org/software/gettext/",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2206,7 +2206,7 @@
     },
     "gettext": {
       "baseline": "0.21",
-      "port-version": 1
+      "port-version": 2
     },
     "gettimeofday": {
       "baseline": "2017-10-14-3",

--- a/versions/g-/gettext.json
+++ b/versions/g-/gettext.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7aba831bc44964ac3d3767392b037e30b06d897e",
+      "version": "0.21",
+      "port-version": 2
+    },
+    {
       "git-tree": "9bb7007478939f7d50c5cd0ca6154c3593c18232",
       "version": "0.21",
       "port-version": 1

